contracts/Strategy.sol

```diff
@@ -274,60 +371,317 @@ contract Strategy is BaseStrategy {
+            totalRatio += _ratios[i];
+        }
+
+        require(totalRatio == 10_000); //ratios must add to 10000 bps
+
+        uint256 wantBalance = balanceOfWant();
+        if (wantBalance > dustThreshold) {
+            bentoBox.deposit(
                 BIERC20(address(want)),
                 address(this),
-                address(kashiPair),
-                sharesInBento
+                address(this),
+                wantBalance,
+                0 // setting this to 0, let's the previous argument determine the deposit size
+            );
+        }
+
+        uint256 totalAssets = estimatedTotalAssets();
+        uint256[] memory kashiPairsIncreasedAllocation =
+            new uint256[](kashiPairs.length);
+
+        for (uint256 i = 0; i < kashiPairs.length; i++) {
+            uint256 pairTotalAssets =
+                bentoSharesToWant(
+                    kashiFractionToBentoShares(
+                        kashiPairs[i].kashiPair,
+                        kashiFractionTotal(i),
+                        true
+                    ),
+                    true
+                );
+            uint256 targetAssets = (_ratios[i] * totalAssets) / 10**4;
+            if (targetAssets < pairTotalAssets) {
+                uint256 toLiquidate = pairTotalAssets.sub(targetAssets);
+                liquidateKashiPair(i, wantToBentoShares(toLiquidate, true));
+            } else if (targetAssets > pairTotalAssets) {
+                kashiPairsIncreasedAllocation[i] = targetAssets.sub(
+                    pairTotalAssets
+                );
+            }
+        }
+
+        for (uint256 i = 0; i < kashiPairs.length; i++) {
+            if (kashiPairsIncreasedAllocation[i] == 0) continue;
+
+            uint256 sharesInBento = sharesInBento();
+            uint256 sharesToAdd =
+                wantToBentoShares(kashiPairsIncreasedAllocation[i], true);
+
+            if (sharesToAdd > sharesInBento) {
+                sharesToAdd = sharesInBento;
+            }
+
+            depositInKashiPair(i, sharesToAdd);
+        }
+    }
+
+    function depositInKashiPair(uint256 kashiPairIndex, uint256 sharesToDeposit)
+        internal
+    {
+        bentoBox.transfer(
+            BIERC20(address(want)),
+            address(this),
+            address(kashiPairs[kashiPairIndex].kashiPair),
+            sharesToDeposit
+        );
+
+        uint256 depositedFraction =
+            kashiPairs[kashiPairIndex].kashiPair.addAsset(
+                address(this),
+                true,
+                sharesToDeposit
+            );
+
+        if (kashiPairs[kashiPairIndex].pid != 0) {
+            masterChef.deposit(
+                kashiPairs[kashiPairIndex].pid,
+                depositedFraction
+            );
+        }
+    }
+
+    function liquidateKashiPair(uint256 kashiPairIndex, uint256 sharesToFree)
+        internal
+        returns (uint256 _shareLiquiduated)
+    {
+        (, uint256 lastAccrued, ) =
```

Comment:
does it make sense to get these .accrue() lines of code outside to an internal function? 

for code reusability + bytecode size savings